File: web/Javascript/Card.js
/*
 * @author Kyle Crossman
 * @file Card.js
 */

//A class to hold the chance cards and community chest cards.
function Card() {
    this.name;
    this.description;
    this.type;
}

//Sets the name.
Card.prototype.setName = function(name) {
    this.name = name;
};

//Sets the description.
Card.prototype.setDescription = function(description) {
    this.description = description;
};

//Sets the type.
Card.prototype.setType = function(type) {
    this.type = type;
};

//Gets the name.
Card.prototype.getName = function() {
    return this.name;
};

//Gets the description.
Card.prototype.getDescription = function() {
    return this.description;
};

//Gets the type.
Card.prototype.getType = function() {
    return this.type;
};

//Converts the card object into a json string.
Card.prototype.toJson = function() {
    var jsonObject = new Object();
    jsonObject.name = this.name;
    jsonObject.description = this.description;
    jsonObject.type = this.type;

    return JSON.stringify(jsonObject);
};

File: src/java/boardObjects/Card.java
package boardObjects;

/**
 * A simple object used to hold the placeholder info about the cards.
 * @author Kyle
 */
public class Card {
    private String name;
    private String description;
    private String type;
    
    //Used for the HTML part. Tells the user what each card does.
    private String details;
    
    public Card() {
        this.name = "";
        this.description = "";
        this.type = "";
        this.details = "";
    }
    
    public Card(String name, String description, String type) {
        this.name = name;
        this.description = description;
        this.type = type;
        this.details = "";
    }

    public Card(String name, String description, String type, String details) {
        this.name = name;
        this.description = description;
        this.type = type;
        this.details = details;
    }
    
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }
}
